rules/DeadCode/Rector/ClassMethod/RemoveUnusedPrivateMethodRector.php

```diff
@@ -67,8 +80,25 @@ public function refactor(Node $node): ?Node
             return null;
         }
 
-        $calls = $this->nodeRepository->findCallsByClassMethod($node);
-        if ($calls !== []) {
+        $class = $node->getAttribute(AttributeKey::CLASS_NODE);
+        if (! $class instanceof Class_) {
+            return null;
+        }
+
+        $classMethodName = $this->nodeNameResolver->getName($node);
+
+        // 1. direct normal calls
+        if ($this->isClassMethodCalledInLocalMethodCall($class, $classMethodName)) {
+            return null;
+        }
+
+        // 2. direct tatic calls
```

Comment:
typo: `tatic` -> `static`: 

```suggestion
        // 2. direct static calls
```